src/ir/auth_logic/ast.h

```diff
@@ -148,26 +179,31 @@ class BaseFact {
     return visitor.Visit(*this);
   }
 
-  // A potentially ugly print of the state in this class
-  // for debugging/testing only
-  std::string DebugPrint() const {
-    return absl::StrCat(
-        "BaseFact(",
-        std::visit([](auto& obj) { return obj.DebugPrint(); }, this->value_),
-        ")");
+  bool operator==(const BaseFact& rhs) const {
+    return value_ == rhs.GetValue();
   }
+  virtual bool operator==(const ASTNode& rhs) const override {
+    try {
+      const BaseFact & rhsBaseFact = dynamic_cast<const BaseFact&>(rhs);
+      return *this == rhsBaseFact;
+    } catch(const std::bad_cast& e) {
+      return false;
+    }
```

Comment:
Don't use a `try...catch` for this, as mentioned in another comment.

Reply:
fixed